modules/mesh-task/variables.tf

```diff
@@ -207,3 +232,115 @@ variable "application_shutdown_delay_seconds" {
   EOT
   default     = null
 }
+
+variable "consul_ecs_config" {
+  type        = any
+  default     = {}
+  description = <<EOT
+  Additional configuration to pass to the consul-ecs binary for Consul service and sidecar proxy registration
+  requests.
+
+  This accepts a subset of the [consul-ecs config file](https://github.com/hashicorp/consul-ecs/blob/main/config/schema.json).
+  For the remainder of the consul-ecs config file contents, use the variables `upstreams`, `checks`, `consul_service_name`,
+  `consul_service_tags`, and `consul_service_meta`. In most cases, these separate variables should suffice.
```

Comment:
This `consul_ecs_config` variable only includes the _additional_ fields we don't already support as dedicated Terraform variables. For example, we already have an `upstreams` variable, so I've excluded `service.proxy.upstreams` from the `consul_ecs_config` variable. Users would continue using the `upstreams` variable to pass in the upstreams.

This works. I personally wouldn't mind deprecating/removing some of the dedicated Terraform variables, in favor of passing them in as part of the config file. In particular, `consul_service_tags`, `consul_service_meta`, and `consul_service_name` could be removed and in favor of using this config variable to pass those. Thoughts?

Reply:
IMO we should either go all in on either using individual variables for everything or the ``consul_ecs_config` variable for everything. Overall, I think using the `consul_ecs_config` variable for everything is better since it matches the interface for consul-ecs. This is a bunch larger breaking change though. What do you think?

Reply:
I'm not sure. The UX is matters a bit here. I was leaning toward keeping Terraform variables for commonly-used fields, and falling back on `consul_ecs_config` for less commonly-used fields. That way there's a bit less nesting, we can add nice TF documentation for those variables, and such.

@lkysow Thoughts?

Reply:
If we go in that direction, we would probably want to consider which fields should have explicit variables rather than just relying on the ones that already have variables.

Reply:
I like offering the variables for things most people will interact with and the `consul_ecs_config` for power users or overrides.

Reply:
> consider which fields should have explicit variables

Right. These are the variables we have currently from the consul-ecs config file:

| consul-ecs config  | terraform variable |
| ------------- | ------------- |
| `service.port`  | `port`  |
| `service.name`  | `consul_service_name`  |
| `service.tags` | `consul_service_tags` |
| `service.meta` | `consul_service_meta` |
| `service.checks` | `checks` |
| `proxy.upstreams` | `upstreams` | 

Do we want to change any of these? I suggested moving the name, tags, and meta into `consul_ecs_config`. But at least tags and meta were added based on user requests.

And we could add a TF variable for `namespace`?

Reply:
yeah consul_namespace would be fine